### DISCUSSION TITLE
What are the recommended browserslist settings?

### DISCUSSION THREAD

**Discussion by muuvmuuv** [tauri-apps/tauri#8785] [Q&A] [answered]
Since we deploy to several environments, we would like to optimize our Tauri environment to not include all polyfills and reduce the bundle size. For that, we currently use browserslist in Angular.

At the moment it looks like this for Web/PWA and Mobile:

```json
{
  "production": [
    "last 10 Chrome versions",
    "last 20 ChromeAndroid versions",
    "last 5 Safari versions",
    "last 3 iOS versions",
    "last 10 Firefox versions",
    "last 10 FirefoxAndroid versions",
    "last 5 Edge versions",
    "Firefox ESR",
    "not dead",
    "> 10% in DE"
  ],
  "mobile": [
    "iOS >= 13",
    "Android >= 10",
    "ChromeAndroid >= 79",
    "Chrome >= 79"
  ]
}
```

What would be the recommendation or supported platform range for Tauri? Since it is WebKit under the hood, what is the minimum compiled version?

**Comment by JonasKruckenberg** [comment, marked as answer]
That kind of depends on what WebKit features your app absolutely can't live without. E.g. some printing related APIs are only available on Big Sur and newer.

You can use [this table here](https://tauri.app/v1/references/webview-versions) to figure out the macOS version that corresponds to the Safari Version you want to support.

So it's less about what version you *must* support and rather what versions you *want* to support.

Though generally speaking here's the browserlist that comes as part of the default template and that will produce code compatible with all windows, most linux and also practically all macOS versions, i.e. a good baseline:

```
[
   'chrome115', // windows
   'safari13' // macOS Mojave and newer + Debian 9 era linux
]
```

A few things of note:
- Windows generally updates the webview automatically and independently which means its sort of evergreen. In theory you could therefore set this to latest chrome, but in practice you want some leeway here for stragglers that for whatever reason didn't update yet.
- The macOS and Linux versions highly depend on which OS versions you want to support. It might make sense for you to bump this version number up a bit if you don't care about pre-BigSur (which you likely don't)
- Since you're building your app and frontend specifically for each OS it makes sense to set the browserlist dynamically based on target. So e.g. you can take advantage of the evergreen chrome on windows. You can use [the env vars we set for the `beforeBuildCommand`](https://tauri.app/v1/api/config) to do so

**Reply by muuvmuuv** [reply]
Wow, thanks for that detailed answer and additional hints. Good point to use different lists for OS builds. Since we only want to support 3 major releases back on macOS and one on Windows, this will work perfectly fine.
